arches/app/media/js/viewmodels/workflow.js

```diff
@@ -288,15 +275,18 @@ define([
             if (furthestValidStepIndex !== self.furthestValidStepIndex()) {
                 self.furthestValidStepIndex(furthestValidStepIndex);
             }
-        });
+    });
 
-        this.updateStepPath = function() {
+        this.updateStepPath = async function() {
+            // Fetch step data in one query, rather than per step
```

Comment:
Before settling on the current approach, I think I noticed like 45 or so GET requests going out when starting the new project workflow in afs. Now it's down to 1.

There are still n POST requests that go out (1 per step), but we don't await them on the frontend, so it's not a big deal.